signup/api/users.py

```diff
@@ -66,13 +66,16 @@ class PasswordChangeAPIView(UpdateAPIView):
 
     def perform_update(self, serializer):
         password = serializer.validated_data['password']
-        serializer.instance.set_password(password)
-        serializer.instance.save()
-        # Updating the password logs out all other sessions for the user
-        # except the current one if
-        # django.contrib.auth.middleware.SessionAuthenticationMiddleware
-        # is enabled.
-        update_session_auth_hash(self.request, serializer.instance)
+        new_password = serializer.validated_data.get('new_password')
+        pwd_correct = self.request.user.check_password(password)
+        if pwd_correct and new_password:
+            serializer.instance.set_password(new_password)
+            serializer.instance.save()
+            # Updating the password logs out all other sessions for the user
+            # except the current one if
+            # django.contrib.auth.middleware.SessionAuthenticationMiddleware
+            # is enabled.
+            update_session_auth_hash(self.request, serializer.instance)
```

Comment:
Should this statement only executed when changing the password for `request.user` instead of all users?

Reply:
Not sure I get this one.

Reply:
The code and document for `[update_session_auth_hash](https://github.com/django/django/blob/aa5fd84f53f09338d01a3cfd9fa6ab08e418fe00/django/contrib/auth/__init__.py#L211)` shows that it should only apply to changing the password of the `request.user`, not any lambda user as the API allows.

Reply:
I submitted a commit where I replaced `serializer.instance` with `self.request.user` -- I hope I understood this correctly.